pkg/policyfilter/state.go

```diff
@@ -205,7 +207,7 @@ func (pol *policy) podMatches(podNs string, podLabels labels.Labels) bool {
 		return false
 	}
 
-	return pol.podSelector.Match(podLabels)
+	return pol.podSelector.Match(podLabels, podNs)
```

Comment:
How about:

```
labels := podLabels
if _, ok  := labels[k8sPodNamespace]; !ok { // only define the namespace label if one does not exist
        labels[k8sPodNamespace] = podNs
}
```

Please add a test for the case where k8sPodNamspace is already a label in the pod.